src/v/cluster/tm_stm_types.h

```diff
@@ -92,7 +182,186 @@ struct tx_metadata {
     std::chrono::milliseconds get_timeout() const { return timeout_ms; }
 
     bool delete_partition(const tx_partition& part);
+    /**
+     * Validates and updates the transaction metadata status if transition is
+     * allowed by transaction FSM.
+     */
+    std::optional<state_transition_error> try_update_status(tx_status);
```

Comment:
should we use it in update to tombstone as well (to check the previous state makes sense)?